.github/workflows/node.yml

```diff
@@ -10,20 +10,39 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
+
     - name: Set up Python 3.7
       uses: actions/setup-python@v4
       with:
         python-version: 3.7
+
+    - name: Install pipenv
+      run: |
+        python -m pip install --upgrade pipenv wheel
+
+    - id: cache-pipenv
+      uses: actions/cache@v1
+      with:
+        path: ~/.local/share/virtualenvs
+        key: ${{ runner.os }}-pipenv-${{ hashFiles('**/Pipfile.lock') }}
+
+    - name: Install dependencies
+      if: steps.cache-pipenv.outputs.cache-hit != 'true'
+      run: |
+        pipenv install --dev
+
     - name: Set up Node
       uses: actions/setup-node@v3
       with:
         # Should be same version as in scripts/bootstrap.sh
         node-version: '16.14.0'
```

Comment:
Since this workflow now requires `pipenv`, arguably we could drop this and change the above `pipenv install --dev` to just run `./scripts/bootstrap.sh`. Then `node`, `yarn` and the node dependencies are installed consistently without any duplication here. Actually, all the workflows should probably be changed to run `bootstrap.sh` so that all dev & test environments are consistent. Maybe another day.